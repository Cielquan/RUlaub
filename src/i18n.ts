import { i18n } from "@lingui/core";
import { de, en } from "make-plural/plurals";
import { messages as deMessages } from "./locales/de-DE/messages";
import { messages as enMessages } from "./locales/en-US/messages";

export type SupportedLocales = "de-DE" | "en-US";
export type SupportedLocalesShorted = "deDE" | "enUS";

export interface Language {
  name: string;
  locale: SupportedLocales;
  importName: SupportedLocalesShorted;
}

export const Languages: { [key: string]: Language } = {
  german: {
    name: "Deutsch",
    locale: "de-DE",
    importName: "deDE",
  },
  english: {
    name: "English",
    locale: "en-US",
    importName: "enUS",
  },
};

export const defaultLocale: SupportedLocales = "en-US";
export const defaultLanguage: Language = Object.values(Languages).filter(
  (lang) => lang.locale === defaultLocale
)[0];

i18n.loadLocaleData({
  "de-DE": { plurals: de },
  "en-US": { plurals: en },
});
i18n.load("de-DE", deMessages);
i18n.load("en-US", enMessages);
i18n.activate(defaultLocale);

export default Languages;
