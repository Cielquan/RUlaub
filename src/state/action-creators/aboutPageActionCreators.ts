import { Dispatch } from "redux";

import { AboutPageActionType } from "../action-types";
import { AboutPageAction } from "../actions";

export const openAboutPageAction = (): AboutPageAction => ({
  type: AboutPageActionType.OPEN,
});

export const openAboutPage =
  () =>
  (dispatch: Dispatch<AboutPageAction>): void => {
    dispatch(openAboutPageAction());
  };

export const closeAboutPageAction = (): AboutPageAction => ({
  type: AboutPageActionType.CLOSE,
});

export const closeAboutPage =
  () =>
  (dispatch: Dispatch<AboutPageAction>): void => {
    dispatch(closeAboutPageAction());
  };
