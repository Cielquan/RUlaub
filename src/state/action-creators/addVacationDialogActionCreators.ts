import { Dispatch } from "redux";

import { AddVacationDialogActionType } from "../action-types";
import { AddVacationDialogAction } from "../actions";

export const openAddVacationDialogAction = (): AddVacationDialogAction => ({
  type: AddVacationDialogActionType.OPEN,
});

export const openAddVacationDialog =
  () =>
  (dispatch: Dispatch<AddVacationDialogAction>): void => {
    dispatch(openAddVacationDialogAction());
  };

export const closeAddVacationDialogAction = (): AddVacationDialogAction => ({
  type: AddVacationDialogActionType.CLOSE,
});

export const closeAddVacationDialog =
  () =>
  (dispatch: Dispatch<AddVacationDialogAction>): void => {
    dispatch(closeAddVacationDialogAction());
  };
