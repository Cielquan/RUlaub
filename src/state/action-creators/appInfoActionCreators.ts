import { Dispatch } from "redux";
import { AppInfoType } from "../action-types";
import { AppInfoAction } from "../actions";

export const changeAppInfoName =
  (newName: string) =>
  (dispatch: Dispatch<AppInfoAction>): void => {
    dispatch({
      type: AppInfoType.NAME,
      payload: newName,
    });
  };

export const changeAppInfoVersion =
  (newVersion: string) =>
  (dispatch: Dispatch<AppInfoAction>): void => {
    dispatch({
      type: AppInfoType.VERSION,
      payload: newVersion,
    });
  };

export const changeAppInfoInfoText =
  (newInfoText: string) =>
  (dispatch: Dispatch<AppInfoAction>): void => {
    dispatch({
      type: AppInfoType.INFO_TEXT,
      payload: newInfoText,
    });
  };
