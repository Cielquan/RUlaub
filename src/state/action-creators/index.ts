export * from "./configActionCreators";
export * from "./infoPageActionCreators";
export * from "./publicHolidaysDataActionCreators";
export * from "./publicHolidaysDialogActionCreators";
export * from "./schoolHolidaysDataActionCreators";
export * from "./settingsDialogActionCreators";
export * from "./sideMenuActionCreators";
export * from "./vacationDataActionCreators";
