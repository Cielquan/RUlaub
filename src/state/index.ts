export * as actionCreators from "./action-creators";
export { default as store } from "./store";
export * from "./reducers";
