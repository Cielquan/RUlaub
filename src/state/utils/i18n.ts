export type SupportedLocales = "de-DE" | "en-US";
export type SupportedLocalesShorted = "deDE" | "enUS";

export interface Language {
  name: string;
  locale: SupportedLocales;
  importName: SupportedLocalesShorted;
}

export const Languages: { [key: string]: Language } = {
  german: {
    name: "Deutsch",
    locale: "de-DE",
    importName: "deDE",
  },
  english: {
    name: "English",
    locale: "en-US",
    importName: "enUS",
  },
};

export default Languages;
