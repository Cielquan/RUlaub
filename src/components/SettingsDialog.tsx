import { t } from "@lingui/macro";
import {
  KeyboardArrowDown as KeyboardArrowDownIcon,
  KeyboardArrowUp as KeyboardArrowUpIcon,
  Settings as SettingsIcon,
} from "@mui/icons-material";
import {
  Button,
  Checkbox,
  Collapse,
  Dialog,
  DialogActions,
  DialogContent,
  DialogTitle,
  FormControl,
  FormControlLabel,
  FormLabel,
  Slide,
  Slider,
  TextField,
} from "@mui/material";
import { TransitionProps } from "@mui/material/transitions";
import { Box } from "@mui/system";
import { invoke } from "@tauri-apps/api";
import { useSnackbar } from "notistack";
import React, { ReactElement, forwardRef, useEffect, useState } from "react";
import { useDispatch, useSelector } from "react-redux";
import { bindActionCreators } from "redux";

import { LogLevel } from "../backendAPI/types/configFile.schema";
import { useAsync } from "../hooks";
import { State, actionCreators } from "../state";

const Transition = forwardRef(
  (
    props: TransitionProps & { children: ReactElement },
    ref: React.Ref<unknown>
    // eslint-disable-next-line react/jsx-props-no-spreading
  ) => <Slide direction="up" ref={ref} {...props} />
);

interface Props {
  onClick?(): void;
}

const SettingsDialog = ({ onClick }: Props): ReactElement => {
  const dispatch = useDispatch();
  const { closeSettingsDialog, setLogLevel, setUserName, setYearChangeScrollBegin } =
    bindActionCreators(actionCreators, dispatch);
  const settingsDialogState = useSelector((state: State) => state.settingsDialog);
  const configState = useSelector((state: State) => state.config);
  // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
  const { settings } = configState!;

  const [name, setName] = useState(configState?.user?.name ?? null);
  const [nameForm, setNameForm] = useState(name);
  const [showAdvanced, setShowAdvanced] = useState(false);
  const [level, setLevel] = useState(settings.logLevel);
  const [levelForm, setLevelForm] = useState(level);
  const [scroll, setScroll] = useState(settings.yearChangeScrollBegin);
  const [scrollForm, setScrollForm] = useState(scroll);

  useEffect(() => {
    setName(configState?.user?.name ?? "");
    setNameForm(configState?.user?.name ?? "");
    setLevel(settings.logLevel);
    setLevelForm(settings.logLevel);
    setShowAdvanced(false);
    setScroll(settings.yearChangeScrollBegin);
    setScrollForm(settings.yearChangeScrollBegin);
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [settingsDialogState]);

  const id = "settings-dialog";

  const snackbarHandles = useSnackbar();

  const { value: availableLogLevels } = useAsync(
    async (): Promise<LogLevel[]> => invoke("get_available_log_levels")
  );
  if (availableLogLevels === undefined) return <></>;

  const marks = availableLogLevels.map((lvl, index) => ({
    value: index,
    label: lvl.charAt(0).toUpperCase() + lvl.slice(1).toLowerCase(),
  }));

  const saveChanges = (): boolean => {
    if (level !== levelForm) {
      setLevel(levelForm);
      setLogLevel(levelForm, snackbarHandles);
    }
    if (scroll !== scrollForm) {
      setScroll(scrollForm);
      setYearChangeScrollBegin(scrollForm, snackbarHandles);
    }
    if (name !== nameForm && nameForm !== undefined && nameForm !== null) {
      setName(nameForm);
      setUserName(nameForm, snackbarHandles);
    }
    return true;
  };

  return (
    <Dialog
      aria-labelledby={id}
      data-testid={id}
      open={settingsDialogState}
      TransitionComponent={Transition}
      keepMounted
      onClose={closeSettingsDialog}
    >
      <DialogTitle id={id} sx={{ display: "flex", alignItems: "center" }}>
        <Box sx={{ flexGrow: 1 }}>{t`Settings`}</Box>
        <SettingsIcon />
      </DialogTitle>
      <DialogContent sx={{ display: "flex", flexDirection: "column" }}>
        <TextField
          autoFocus
          margin="dense"
          id="name"
          label={t`Name`}
          type="text"
          fullWidth
          variant="outlined"
          value={nameForm}
          helperText={t`Enter your name to have it be
          preselected when entering new vacation.`}
          onChange={(event): void => {
            setNameForm(event.target.value);
          }}
          sx={{ marginY: 1 }}
        />
        <Box
          sx={{
            marginY: 2,
            display: "flex",
            justifyContent: "center",
            alignItems: "center",
            "&::before, &::after": {
              content: "''",
              borderBottom: "solid 2px",
              borderColor: "primary.main",
              flexGrow: 1,
              marginX: 1,
            },
          }}
        >
          <Button
            onClick={() => setShowAdvanced(!showAdvanced)}
            endIcon={showAdvanced ? <KeyboardArrowUpIcon /> : <KeyboardArrowDownIcon />}
          >
            {t`Advanced Settings`}
          </Button>
        </Box>
        <Collapse in={showAdvanced}>
          <FormControl
            component="fieldset"
            fullWidth
            sx={{
              marginY: 1,
              padding: 1,
              border: 1,
              borderRadius: 1,
              borderColor: "action.disabled",
              "&:hover": {
                borderColor: "action.active",
              },
            }}
          >
            <FormLabel component="legend" sx={{ paddingX: 1, fontSize: "0.75em" }}>
              {t`Logging Level`}
            </FormLabel>
            <Box sx={{ width: "90%", marginX: "5%" }}>
              <Slider
                key={`log-level-slider-at-${availableLogLevels.indexOf(levelForm)}`}
                marks={marks}
                defaultValue={availableLogLevels.indexOf(levelForm)}
                min={0}
                max={marks.length - 1}
                onChangeCommitted={(event, newValue) => {
                  setLevelForm(availableLogLevels[newValue as number]);
                }}
              />
            </Box>
          </FormControl>
          <FormControl
            component="fieldset"
            fullWidth
            sx={{
              marginY: 1,
              padding: 1,
              border: 1,
              borderRadius: 1,
              borderColor: "action.disabled",
              "&:hover": {
                borderColor: "action.active",
              },
            }}
          >
            <FormControlLabel
              label={t`On year switch, scroll to beginning
              of the year, if not current year.`}
              control={
                <Checkbox
                  checked={scrollForm}
                  onChange={(event) => setScrollForm(event.target.checked)}
                />
              }
            />
          </FormControl>
        </Collapse>
      </DialogContent>
      <DialogActions>
        <Button
          data-testid={`${id}-btn-save`}
          onClick={() => {
            if (typeof onClick === "function") onClick();
            if (!saveChanges()) return;
            closeSettingsDialog();
          }}
          autoFocus
        >
          {t`Save`}
        </Button>
        <Button
          data-testid={`${id}-btn-cancel`}
          onClick={() => {
            if (typeof onClick === "function") onClick();
            closeSettingsDialog();
          }}
          autoFocus
        >
          {t`Discard`}
        </Button>
      </DialogActions>
    </Dialog>
  );
};
SettingsDialog.defaultProps = {
  onClick: () => undefined,
};

export default SettingsDialog;
