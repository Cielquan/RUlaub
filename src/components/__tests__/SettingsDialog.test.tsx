import { I18nProvider } from "@lingui/react";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import React from "react";
import { Provider } from "react-redux";
import { CombinedState } from "redux";
import { MockStoreEnhanced } from "redux-mock-store";

import i18n from "../../i18n";
import { State, initialState } from "../../state";
import { closeSettingsDialogAction, updateConfigAction } from "../../state/action-creators";
import generateMockStore from "../../testUtils";
import SettingsDialog from "../SettingsDialog";

describe("<SettingsDialog />", () => {
  let testState: State;
  let mockStore: MockStoreEnhanced<Partial<CombinedState<State>>>;

  beforeEach(() => {
    testState = JSON.parse(JSON.stringify(initialState));
    mockStore = generateMockStore(testState);
  });

  it("renders invisible by default", () => {
    expect(mockStore.getState().aboutPage).toBe(false);
    render(
      <Provider store={mockStore}>
        <I18nProvider i18n={i18n}>
          <SettingsDialog />
        </I18nProvider>
      </Provider>
    );
    expect(screen.getByTestId("settings-dialog")).toMatchSnapshot();
    expect(screen.getByTestId("settings-dialog")).not.toBeVisible();
  });

  it("renders correctly when open", () => {
    testState.settingsDialog = true;
    const alteredMockStore = generateMockStore(testState);
    expect(alteredMockStore.getState().settingsDialog).toBe(true);
    render(
      <Provider store={mockStore}>
        <I18nProvider i18n={i18n}>
          <SettingsDialog />
        </I18nProvider>
      </Provider>
    );
    expect(screen.getByTestId("settings-dialog")).toMatchSnapshot();
    expect(screen.getByTestId("settings-dialog")).toBeVisible();
  });

  it("closes when button is clicked", () => {
    testState.settingsDialog = true;
    const alteredMockStore = generateMockStore(testState);
    const onClickMock = jest.fn();

    expect(alteredMockStore.getState().settingsDialog).toBe(true);
    render(
      <Provider store={alteredMockStore}>
        <I18nProvider i18n={i18n}>
          <SettingsDialog onClick={onClickMock} />
        </I18nProvider>
      </Provider>
    );
    expect(screen.getByTestId("settings-dialog")).toBeVisible();

    userEvent.click(screen.getByTestId("settings-dialog-btn"));
    expect(onClickMock).toHaveBeenCalledTimes(1);
    const expectedActions = [
      closeSettingsDialogAction(),
      updateConfigAction({ user: { name: undefined } }),
    ];
    expect(alteredMockStore.getActions()).toEqual(expectedActions);
  });
});
