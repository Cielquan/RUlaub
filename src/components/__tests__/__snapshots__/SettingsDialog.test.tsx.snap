// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SettingsDialog /> renders correctly when open 1`] = `
<div
  class="MuiModal-root MuiDialog-root css-zw3mfo-MuiModal-root-MuiDialog-root"
  data-testid="settings-dialog"
  role="presentation"
>
  <div
    aria-hidden="true"
    class="MuiBackdrop-root css-yiavyu-MuiBackdrop-root-MuiDialog-backdrop"
    style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
  />
  <div
    data-test="sentinelStart"
    tabindex="0"
  />
  <div
    class="MuiDialog-container MuiDialog-scrollPaper css-hz1bth-MuiDialog-container"
    role="presentation"
    style="webkit-transform: none; transform: none; webkit-transition: -webkit-transform 225ms cubic-bezier(0.0, 0, 0.2, 1) 0ms; transition: transform 225ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;"
    tabindex="-1"
  >
    <div
      aria-labelledby="settings-dialog"
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation24 MuiDialog-paper MuiDialog-paperScrollPaper MuiDialog-paperWidthSm css-1t1j96h-MuiPaper-root-MuiDialog-paper"
      role="dialog"
    >
      <h2
        class="MuiTypography-root MuiTypography-h6 MuiDialogTitle-root css-tqj0yt-MuiTypography-root-MuiDialogTitle-root"
        id="settings-dialog"
      >
        <div
          class="MuiBox-root css-i9gxme"
        >
          Settings
        </div>
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="SettingsIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19.14 12.94c.04-.3.06-.61.06-.94 0-.32-.02-.64-.07-.94l2.03-1.58c.18-.14.23-.41.12-.61l-1.92-3.32c-.12-.22-.37-.29-.59-.22l-2.39.96c-.5-.38-1.03-.7-1.62-.94l-.36-2.54c-.04-.24-.24-.41-.48-.41h-3.84c-.24 0-.43.17-.47.41l-.36 2.54c-.59.24-1.13.57-1.62.94l-2.39-.96c-.22-.08-.47 0-.59.22L2.74 8.87c-.12.21-.08.47.12.61l2.03 1.58c-.05.3-.09.63-.09.94s.02.64.07.94l-2.03 1.58c-.18.14-.23.41-.12.61l1.92 3.32c.12.22.37.29.59.22l2.39-.96c.5.38 1.03.7 1.62.94l.36 2.54c.05.24.24.41.48.41h3.84c.24 0 .44-.17.47-.41l.36-2.54c.59-.24 1.13-.56 1.62-.94l2.39.96c.22.08.47 0 .59-.22l1.92-3.32c.12-.22.07-.47-.12-.61l-2.01-1.58zM12 15.6c-1.98 0-3.6-1.62-3.6-3.6s1.62-3.6 3.6-3.6 3.6 1.62 3.6 3.6-1.62 3.6-3.6 3.6z"
          />
        </svg>
      </h2>
      <div
        class="MuiDialogContent-root css-ypiqx9-MuiDialogContent-root"
      >
        <p
          class="MuiDialogContentText-root MuiTypography-root MuiTypography-body1 css-e901ri-MuiTypography-root-MuiDialogContentText-root"
        >
          <span
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            Enter your name to have it be preselected when entering new vacation.
          </span>
        </p>
        <div
          class="MuiFormControl-root MuiFormControl-marginDense MuiFormControl-fullWidth MuiTextField-root css-1z10yd4-MuiFormControl-root-MuiTextField-root"
        >
          <label
            class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard MuiFormLabel-root MuiFormLabel-colorPrimary css-aqpgxn-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="false"
            for="name"
            id="name-label"
          >
            Name
          </label>
          <div
            class="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-ghsjzk-MuiInputBase-root-MuiInput-root"
          >
            <input
              aria-invalid="false"
              class="MuiInput-input MuiInputBase-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
              id="name"
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
      <div
        class="MuiDialogActions-root MuiDialogActions-spacing css-hlj6pa-MuiDialogActions-root"
      >
        <button
          class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root Mui-focusVisible css-1e6y48t-MuiButtonBase-root-MuiButton-root"
          data-testid="settings-dialog-btn"
          tabindex="0"
          type="button"
        >
          Save
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          >
            <span
              class="css-y4cjyz-MuiTouchRipple-ripple MuiTouchRipple-ripple MuiTouchRipple-rippleVisible MuiTouchRipple-ripplePulsate"
              style="width: 1px; height: 1px; top: -0.5px; left: -0.5px;"
            >
              <span
                class="MuiTouchRipple-child MuiTouchRipple-childPulsate"
              />
            </span>
          </span>
        </button>
      </div>
    </div>
  </div>
  <div
    data-test="sentinelEnd"
    tabindex="0"
  />
</div>
`;

exports[`<SettingsDialog /> renders invisible by default 1`] = `
<div
  aria-hidden="true"
  class="MuiModal-root MuiModal-hidden MuiDialog-root css-ew73zq-MuiModal-root-MuiDialog-root"
  data-testid="settings-dialog"
  role="presentation"
>
  <div
    aria-hidden="true"
    class="MuiBackdrop-root css-yiavyu-MuiBackdrop-root-MuiDialog-backdrop"
    style="opacity: 0; visibility: hidden;"
  />
  <div
    data-test="sentinelStart"
    tabindex="0"
  />
  <div
    class="MuiDialog-container MuiDialog-scrollPaper css-hz1bth-MuiDialog-container"
    role="presentation"
    style="visibility: hidden; webkit-transform: translateY(768px); transform: translateY(768px);"
    tabindex="-1"
  >
    <div
      aria-labelledby="settings-dialog"
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation24 MuiDialog-paper MuiDialog-paperScrollPaper MuiDialog-paperWidthSm css-1t1j96h-MuiPaper-root-MuiDialog-paper"
      role="dialog"
    >
      <h2
        class="MuiTypography-root MuiTypography-h6 MuiDialogTitle-root css-tqj0yt-MuiTypography-root-MuiDialogTitle-root"
        id="settings-dialog"
      >
        <div
          class="MuiBox-root css-i9gxme"
        >
          Settings
        </div>
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="SettingsIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19.14 12.94c.04-.3.06-.61.06-.94 0-.32-.02-.64-.07-.94l2.03-1.58c.18-.14.23-.41.12-.61l-1.92-3.32c-.12-.22-.37-.29-.59-.22l-2.39.96c-.5-.38-1.03-.7-1.62-.94l-.36-2.54c-.04-.24-.24-.41-.48-.41h-3.84c-.24 0-.43.17-.47.41l-.36 2.54c-.59.24-1.13.57-1.62.94l-2.39-.96c-.22-.08-.47 0-.59.22L2.74 8.87c-.12.21-.08.47.12.61l2.03 1.58c-.05.3-.09.63-.09.94s.02.64.07.94l-2.03 1.58c-.18.14-.23.41-.12.61l1.92 3.32c.12.22.37.29.59.22l2.39-.96c.5.38 1.03.7 1.62.94l.36 2.54c.05.24.24.41.48.41h3.84c.24 0 .44-.17.47-.41l.36-2.54c.59-.24 1.13-.56 1.62-.94l2.39.96c.22.08.47 0 .59-.22l1.92-3.32c.12-.22.07-.47-.12-.61l-2.01-1.58zM12 15.6c-1.98 0-3.6-1.62-3.6-3.6s1.62-3.6 3.6-3.6 3.6 1.62 3.6 3.6-1.62 3.6-3.6 3.6z"
          />
        </svg>
      </h2>
      <div
        class="MuiDialogContent-root css-ypiqx9-MuiDialogContent-root"
      >
        <p
          class="MuiDialogContentText-root MuiTypography-root MuiTypography-body1 css-e901ri-MuiTypography-root-MuiDialogContentText-root"
        >
          <span
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            Enter your name to have it be preselected when entering new vacation.
          </span>
        </p>
        <div
          class="MuiFormControl-root MuiFormControl-marginDense MuiFormControl-fullWidth MuiTextField-root css-1z10yd4-MuiFormControl-root-MuiTextField-root"
        >
          <label
            class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard MuiFormLabel-root MuiFormLabel-colorPrimary css-aqpgxn-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="false"
            for="name"
            id="name-label"
          >
            Name
          </label>
          <div
            class="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-ghsjzk-MuiInputBase-root-MuiInput-root"
          >
            <input
              aria-invalid="false"
              class="MuiInput-input MuiInputBase-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
              id="name"
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
      <div
        class="MuiDialogActions-root MuiDialogActions-spacing css-hlj6pa-MuiDialogActions-root"
      >
        <button
          class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root Mui-focusVisible css-1e6y48t-MuiButtonBase-root-MuiButton-root"
          data-testid="settings-dialog-btn"
          tabindex="0"
          type="button"
        >
          Save
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          >
            <span
              class="css-y4cjyz-MuiTouchRipple-ripple MuiTouchRipple-ripple MuiTouchRipple-rippleVisible MuiTouchRipple-ripplePulsate"
              style="width: 1px; height: 1px; top: -0.5px; left: -0.5px;"
            >
              <span
                class="MuiTouchRipple-child MuiTouchRipple-childPulsate"
              />
            </span>
          </span>
        </button>
      </div>
    </div>
  </div>
  <div
    data-test="sentinelEnd"
    tabindex="0"
  />
</div>
`;
